src/librustdoc/clean/types.rs

```diff
@@ -1306,6 +1306,21 @@ impl Lifetime {
     }
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, Debug, Hash)]
+pub(crate) enum PreciseCapturingArg {
+    Lifetime(Lifetime),
+    Param(Symbol),
+}
+
+impl PreciseCapturingArg {
+    pub(crate) fn name(&self) -> &Symbol {
```

Comment:
NIT: [`Symbol`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_span/symbol/struct.Symbol.html) is `Copy`, so should be returned by value here.